src/main/java/com/tattsun/RaiseTechTask7/UserController.java

```diff
@@ -1,48 +1,48 @@
 package com.tattsun.RaiseTechTask7;
 
-import org.springframework.http.HttpStatus;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 import org.springframework.web.util.UriComponentsBuilder;
 
 import java.net.URI;
 import java.time.LocalDate;
-import java.time.format.DateTimeParseException;
-import java.util.ArrayList;
-import java.util.List;
+import java.util.*;
 
 @RestController
 public class UserController {
-    @GetMapping("/profiles")
-    public ResponseEntity<List<UserProfile>> getUserProfile(
-        @RequestParam(name = "name", required = false) String name,
-        @RequestParam(name = "birthdate", required = false) String stBirthDate
-    ) {
-        List<UserProfile> userProfiles = new ArrayList<>();
-
-        // nameとbirthdateがnullもしくは空文字でない場合、その値を使用してUserProfileを作成
-        if (name != null && !name.isEmpty() && stBirthDate != null && !stBirthDate.isEmpty()) {
-            try {
-                LocalDate birthDate = LocalDate.parse(stBirthDate);
-                userProfiles.add(new UserProfile(name, birthDate));
-            } catch (DateTimeParseException e) {
-                // 無効な日付の文字列が渡された場合のエラーハンドリング
-                return ResponseEntity.badRequest().build();
 
-            }
-        } else {
-            // クエリ文字列が指定されていない場合、ダミーデータを返す
-            userProfiles.add(new UserProfile("Taro Inaba", LocalDate.of(2004, 9, 23)));
-            userProfiles.add(new UserProfile("Ichiro Matsumoto", LocalDate.of(2000, 3, 27)));
-        }
+    Map<Integer,String> UserProfile = new LinkedHashMap<>();
+    {
+        UserProfile.put(1, "Taro Inaba");
+        UserProfile.put(2, "Ichiro Matsumoto");
+        UserProfile.put(3, "Jiro Masuda");
+    }
 
-            return ResponseEntity.ok(userProfiles);
+    @GetMapping("/profiles/{id}")
+    public ResponseEntity<UserProfile> getUserProfile (@PathVariable int id) {
+        String userProfile = UserProfile.get(id);
+            if (userProfile != null) {
+                UserProfile profile = new UserProfile(id, userProfile);
+                return ResponseEntity.ok(profile);
+            } else {
+                return ResponseEntity.notFound().build();
+            }
     }
 
+    @Value("${base.url:http://localhost:8080}")
+    private String baseurl;
+
     @PostMapping("/profiles")
-    public ResponseEntity<CreateResponse> createUserProfile(@RequestBody CreateForm form) {
-        URI url = UriComponentsBuilder.fromUriString("http://localhost:8080")
-                .path("/profiles/1")
+    public ResponseEntity<CreateResponse> createUserProfile(@org.jetbrains.annotations.NotNull @RequestBody  CreateForm form) {
```

Comment:
＠Validアノテーションか＠Validatedアノテーションが必要ですね。
```suggestion
    public ResponseEntity<CreateResponse> createUserProfile(@Valid @RequestBody  CreateForm form) {
```